hedging/src/adjustment_action.rs

```diff
@@ -63,15 +59,18 @@ impl AdjustmentAction {
 pub fn determine_action(
     abs_liability: SyntheticCentLiability,
     signed_exposure: SyntheticCentExposure,
+    HedgingFundingConfig(hedging_config, _): HedgingFundingConfig,
```

Comment:
The difference between config and the other args is its lifecycle. Config doesn't change at runtime - the other arguments do.

This means we want to instantiate a (long living) struct that can keep the config internally:

```
pub struct HedgingAdjustment {
  pub config: HedgingConfig
}

impl HedgingAdjustment {
  pub fn determin_action(...) -> AdjustmentAction
}
```

The same applies to the funding config. Then we don't need to wire the config through all the layers - just the instantiated struct.

Reply:
great thx for that